src/Shouldly/ShouldlyExtensionMethods/ShouldBeDecoratedWithExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Shouldly
+{
+    [DebuggerStepThrough]
+    [ShouldlyMethods]
+    public static class ShouldBeDecoratedWithExtensions
+    {
+        public static void ShouldBeDecoratedWith<T>(this Type actual) where T : class
```

Comment:
```suggestion
        public static void ShouldBeDecoratedWith<T>(this Type actual) where T : Attribute
```

Same for other methods.

Reply:
Updated